src/report.rs

```diff
@@ -0,0 +1,172 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Intel SGX Enclave report structures.
+
+use crate::parameters::{Attributes, Features, MiscSelect, Xfrm};
+
+use core::fmt::{self, Display};
+
+#[derive(Debug, PartialEq)]
+#[non_exhaustive]
+pub enum ReportError {
+    InvalidMiscSelect(u32),
+    InvalidFeatures(u64),
+    InvalidXfrm(u64),
+}
+
+impl Display for ReportError {
```

Comment:
Take a look at the thiserror crate: https://crates.io/crates/thiserror . It's widely used and we already have it inexorably in our dependency graph so we may as well use it ourselves. It's from the same person behind anyhow (also serde), and whereas anyhow is for easily consuming errors (useful for applications), thiserror is for easily defining errors (useful for libraries). As a side effect it can generate Display impls for you.

Reply:
https://github.com/dtolnay/thiserror/pull/64

Reply:
I don't think we need a dependency to define a 3-variant error in a singular location within a crate